### ISSUE TITLE
🚀 Feature: export compat-wrapped configs from the `experimental` api

### ISSUE THREAD

**Issue by michaelfaith** [JoshuaKGoldberg/eslint-plugin-package-json#1865]
### Feature Request Checklist

- [x] I have [searched for related issues](https://github.com/michaelfaith/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

This contributes to Milestone 1 of our [migration plan](https://github.com/michaelfaith/eslint-plugin-package-json/blob/main/docs/plans/parser-to-language-migration/plan.md#milestone-1-experimental-language-support). For Milestone 1, we'll be exporting our existing rules and plugin wrapped in a compatibility layer that allows them to be used with `@eslint/json`.  

This specific change is to update the configs being exported from `experimental` to all be versions of the existing configs with `parser` replaced by a `language` from `@eslint/json`.   

The "Experimental API" doc page should be updated with information about how to use this new version of the configs.

> [!IMPORTANT]
> When we first introduce the dependency on `@eslint/json`, it will have to initially be an optional peer dependency.  
> We can't directly depend on it, until v2.0.0, since it requires at least `eslint` `9.15.0`, whereas our plugin currently supports as low as `9.0.0`.  
> As part of the v2.0.0 release, we'll move `@eslint/json` to be a proper dependency.

**Comment by michaelfaith** [comment]
> When we first introduce the dependency on @eslint/json, it will have to initially be an optional peer dependency.
We can't directly depend on it, until v2.0.0, since it requires at least eslint 9.15.0, whereas our plugin currently supports as low as 9.0.0.
> As part of the v2.0.0 release, we'll move @eslint/json to be a proper dependency.

I double checked the `peerDependencies` in `@eslint/json`, and for some weird reason, it doesn't even have a peer dependency defined for `eslint`.  That seems like a big gap on their side, but that could change how we handle this piece, since adding it as a regular dependency actually wouldn't be a breaking change.  It won't work for versions of eslint less than 9.15.0, but there's not a technical barrier for us to include it as a dependency pre-v2.   Sort of leaning towards just including it as regular dependency as part of this change after all, rather than forcing folks to install it separately during Milestone 1.  @JoshuaKGoldberg do you have a preference?

**Comment by michaelfaith** [comment]
The main downside to including it now, is that all users would now  be downloading it as part of installing their node_modules, even if they're not using the experimental api.

**Comment by michaelfaith** [comment]
Actually... I'm wondering if we should include the plugin at all...  If people are using `@eslint/json` already for its rules, then they'll have already defined `json: jsonPlugin` in their configs, and us redefining it could clash.  Maybe we should declare the language, and treat `@eslint/json` as a peer dependency entirely. 

Which would change this diff from my original plan
```diff
+ import jsonPlugin from '@eslint/json';
+
recommended: {
  files: ['**/package.json'],
- languageOptions: {
-   parser: parserJsonc,
- },
+ language: 'json/json',
  name: 'package-json/recommended',
  plugins: {
+   json: jsonPlugin,
    get 'package-json'(): ESLint.Plugin {
      return plugin;
    },
  },
  rules: recommendedRules,
},
```

To
```diff
recommended: {
  files: ['**/package.json'],
- languageOptions: {
-   parser: parserJsonc,
- },
+ language: 'json/json',
  name: 'package-json/recommended',
  plugins: {
    get 'package-json'(): ESLint.Plugin {
      return plugin;
    },
  },
  rules: recommendedRules,
},
```

**Comment by JoshuaKGoldberg** [comment]
> it doesn't even have a peer dependency defined
> then they'll have already defined

Yeah that's my thought: that there are some package relationships where folks effectively always have both or neither.

I would feel comfortable relying on that and waiting for someone to file a bug. There can even be a docs notice saying _"you have to have it installed; file a bug if you need it"_.

There might be some tooling out the (Nx/Turborepo/esque?) that needs those explicit deps? But since this is linter-tooling-land, not package relationships, I think there's a _chance_ this'll generally be fine.

**Comment by michaelfaith** [comment]
> There might be some tooling out the (Nx/Turborepo/esque?) that needs those explicit deps? 

Oh, so maybe my explanation wasn't clear.  When we remove it from our config, we're not importing anything from the package anymore, and so we don't need it as a dep.  User's _will_ have to import it in order to do the plugin assignment in user-land (since we're not).  Does that make sense.

```ts
import jsonPlugin from '@eslint/json';

// rest of config
plugins: {
   json: jsonPlugin,
}
```

**Comment by michaelfaith** [comment]
Hmm. I just tested it with both us defining the plugin _and_ users defining the plugin themselves, and it doesn't cause issues (I think it used to as some point...).  So maybe we do include it in ours?  Users that are already using it won't be affected, but users that aren't won't have to install something new?

**Comment by michaelfaith** [comment]
Ok, so I got some guidance from the ESLint team, and they're discouraging including the plugin for languages in our configs.  It can create that issue I was afraid of, if our version of the plugin isn't exactly the same as one the user has installed.  So the recommendation is for us to include it as a peer dependency and document that the plugin needs to be defined in user-land.  I'll update the migration plan as part of this change.
